// aurelia-toolbelt/dist/es2015/components/bootstrap/progressbar/abt-progress-bar.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var BootstrapProgressBar = (function () {
    function BootstrapProgressBar() {
        this.value = 0;
        this.min = 0;
        this.max = 100;
        this.animated = false;
        this.striped = false;
        this.isAnimated = false;
        this.isStriped = false;
    }
    BootstrapProgressBar.prototype.afterAttached = function () {
        var animated = (this.animated === '' && this.progressbarTemplate.hasAttribute('animated')) || this.animated.toString() === 'true';
        var striped = (this.striped === '' && this.progressbarTemplate.hasAttribute('striped')) || this.striped.toString() === 'true';
        this.value = Number(this.value);
        this.min = Number(this.min);
        this.max = Number(this.max);
        if (this.color && this.gradientColor) {
            this.gradientColorChanged(this.gradientColor);
        }
    };
    BootstrapProgressBar.prototype.gradientColorChanged = function (newValue) {
        if (this.progressbar) {
            aurelia_framework_1.DOM.injectStyles("\n      #" + this.progressbar.id + "\n      {\n        background: -webkit-gradient(linear, left top, right top, from(" + this.color + "),to(" + newValue + ")) !important;\n        background: -webkit-linear-gradient(left, " + this.color + " 0%," + newValue + " 100%) !important;\n        background: -o-linear-gradient(left, " + this.color + " 0%," + newValue + " 100%) !important;\n        background: linear-gradient(left, " + this.color + " 0%," + newValue + " 100%) !important;\n      }\n      ");
        }
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapProgressBar.prototype, "type", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapProgressBar.prototype, "color", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapProgressBar.prototype, "gradientColor", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapProgressBar.prototype, "style", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapProgressBar.prototype, "class", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], BootstrapProgressBar.prototype, "value", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], BootstrapProgressBar.prototype, "min", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], BootstrapProgressBar.prototype, "max", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], BootstrapProgressBar.prototype, "animated", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], BootstrapProgressBar.prototype, "striped", void 0);
    BootstrapProgressBar = __decorate([
        aurelia_framework_1.customElement('abt-progress-bar'),
        aurelia_framework_1.containerless()
    ], BootstrapProgressBar);
    return BootstrapProgressBar;
}());
exports.BootstrapProgressBar = BootstrapProgressBar;
