// aurelia-toolbelt/dist/es2015/components/bootstrap/carousel/abt-carousel.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var $ = require("jquery");
var CarouselCustomElement = (function () {
    function CarouselCustomElement() {
        this.prevTitle = 'Previous';
        this.nextTitle = 'Next';
        this.prevIcon = 'carousel-control-prev-icon';
        this.nextIcon = 'carousel-control-next-icon';
        this.navigator = false;
        this.indicator = false;
        this.interval = 5000;
        this.keyboard = true;
        this.pause = true;
        this.ride = false;
        this.wrap = true;
        this.showNavigator = false;
        this.showIndicator = false;
    }
    CarouselCustomElement.prototype.afterAttached = function () {
        var _this = this;
        this.interval = Number(this.interval);
        this.showNavigator = (this.navigator === '' && this.carouselTemplate.hasAttribute('navigator')) || this.navigator.toString() === 'true';
        this.showIndicator = (this.indicator === '' && this.carouselTemplate.hasAttribute('indicator')) || this.navigator.toString() === 'true';
        this.keyboard = (this.keyboard === '' && this.carouselTemplate.hasAttribute('keyboard')) || this.keyboard.toString() === 'true';
        this.pause = (this.pause === '' && this.carouselTemplate.hasAttribute('pause')) || this.pause.toString() === 'true';
        this.ride = (this.ride === '' && this.carouselTemplate.hasAttribute('ride')) || this.ride.toString() === 'true';
        this.wrap = (this.wrap === '' && this.carouselTemplate.hasAttribute('wrap')) || this.wrap.toString() === 'true';
        $(this.carousel).carousel({
            interval: this.interval,
            keyboard: this.keyboard,
            pause: this.pause,
            ride: this.ride,
            wrap: this.wrap
        });
        if (this.bsSlide) {
            $(this.carousel).on('slide.bs.carousel', function () {
                if (_this.bsSlide) {
                    _this.bsSlide();
                }
            });
        }
        if (this.bsSlid) {
            $(this.carousel).on('slid.bs.carousel', function () {
                if (_this.bsSlid) {
                    _this.bsSlid();
                }
            });
        }
    };
    CarouselCustomElement.prototype.detached = function () {
        $(this.carousel).carousel('dispose');
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], CarouselCustomElement.prototype, "prevTitle", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], CarouselCustomElement.prototype, "nextTitle", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], CarouselCustomElement.prototype, "prevIcon", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], CarouselCustomElement.prototype, "nextIcon", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "navigator", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "indicator", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "interval", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "keyboard", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "pause", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "ride", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], CarouselCustomElement.prototype, "wrap", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], CarouselCustomElement.prototype, "bsSlide", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], CarouselCustomElement.prototype, "bsSlid", void 0);
    CarouselCustomElement = __decorate([
        aurelia_framework_1.customElement('abt-carousel')
    ], CarouselCustomElement);
    return CarouselCustomElement;
}());
exports.CarouselCustomElement = CarouselCustomElement;
