// aurelia-toolbelt/dist/system/utilities/vanilla/jsTools.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, JsTools;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            JsTools = (function () {
                function JsTools() {
                }
                JsTools.prototype.isString = function (value) {
                    return typeof value === 'string' || value instanceof String;
                };
                JsTools.prototype.isNumber = function (value) {
                    return typeof value === 'number' && isFinite(value);
                };
                JsTools.prototype.isArray = function (value) {
                    return value && typeof value === 'object' && value.constructor === Array;
                };
                JsTools.prototype.isFunction = function (value) {
                    return typeof value === 'function';
                };
                JsTools.prototype.isObject = function (value) {
                    return value && typeof value === 'object' && value.constructor === Object;
                };
                JsTools.prototype.isEmpty = function (value) {
                    for (var key in value) {
                        if (value.hasOwnProperty(key)) {
                            return false;
                        }
                    }
                    return true;
                };
                JsTools.prototype.isNull = function (value) {
                    return value === null;
                };
                JsTools.prototype.isUndefined = function (value) {
                    return typeof value === 'undefined';
                };
                JsTools.prototype.isBoolean = function (value) {
                    return typeof value === 'boolean';
                };
                JsTools.prototype.isRegExp = function (value) {
                    return value && typeof value === 'object' && value.constructor === RegExp;
                };
                JsTools.prototype.isError = function (value) {
                    return value instanceof Error && typeof value.message !== 'undefined';
                };
                JsTools.prototype.isDate = function (value) {
                    return value instanceof Date;
                };
                JsTools.prototype.isSymbol = function (value) {
                    return typeof value === 'symbol';
                };
                JsTools.prototype.jsonFormatter = function (value) {
                    return JSON.stringify(value, null, '\t');
                };
                JsTools.prototype.isUrl = function (value) {
                    if (!this.isString(value)) {
                        return false;
                    }
                    var pattern = new RegExp('^((https?:)?\\/\\/)?' +
                        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' +
                        '((\\d{1,3}\\.){3}\\d{1,3}))' +
                        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' +
                        '(\\?[;&a-z\\d%_.~+=-]*)?' +
                        '(\\#[-a-z\\d_]*)?$', 'i');
                    if (!pattern.test(value)) {
                        return false;
                    }
                    else {
                        return true;
                    }
                };
                JsTools = __decorate([
                    aurelia_framework_1.singleton()
                ], JsTools);
                return JsTools;
            }());
            exports_1("JsTools", JsTools);
        }
    };
});
