// aurelia-toolbelt/dist/system/utilities/vanilla/sharedIndex.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, SharedIndex;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            SharedIndex = (function () {
                function SharedIndex() {
                    this.dictionary = [];
                }
                SharedIndex.prototype.getAndIncrement = function (id) {
                    id = this.getId(id);
                    if (this.dictionary[id] === undefined) {
                        this.dictionary[id] = 0;
                    }
                    else {
                        this.dictionary[id] += 1;
                    }
                    return this.dictionary[id];
                };
                SharedIndex.prototype.getValue = function (id) {
                    id = this.getId(id);
                    return this.dictionary[id];
                };
                SharedIndex.prototype.reset = function (id) {
                    id = this.getId(id);
                    this.dictionary[id] = 0;
                };
                SharedIndex.prototype.resetAll = function () {
                    this.dictionary = [];
                };
                SharedIndex.prototype.getId = function (id) {
                    return 'a' + id.replace(new RegExp('-', 'g'), '');
                };
                SharedIndex = __decorate([
                    aurelia_framework_1.singleton()
                ], SharedIndex);
                return SharedIndex;
            }());
            exports_1("SharedIndex", SharedIndex);
        }
    };
});
