// aurelia-toolbelt/dist/system/components/vanilla/mark-down/aut-mark-down.js
System.register(["aurelia-framework", "aurelia-http-client"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_http_client_1, MarkDownItCustomElement;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_http_client_1_1) {
                aurelia_http_client_1 = aurelia_http_client_1_1;
            }
        ],
        execute: function () {
            MarkDownItCustomElement = (function () {
                function MarkDownItCustomElement(http) {
                    this.http = http;
                    this.isLoading = false;
                    this.myText = '';
                    this.showToolBar = false;
                    this.showPreview = true;
                    this.showEditor = false;
                    this.src = '';
                }
                MarkDownItCustomElement.prototype.afterAttached = function () {
                    this.myText = this.slotContainer.innerHTML.replace('<!--slot-->', '').trim();
                    this.srcChanged(this.src);
                };
                MarkDownItCustomElement.prototype.srcChanged = function (newValue) {
                    var _this = this;
                    this.isLoading = true;
                    return this.http.get(newValue || this.src)
                        .then(function (data) {
                        _this.myText = data.response;
                        _this.isLoading = false;
                    });
                };
                MarkDownItCustomElement.prototype.addText = function (text, cursorPosition) {
                    if (cursorPosition === void 0) { cursorPosition = 0; }
                    if (!this.editor) {
                        return;
                    }
                    var scrollPos = this.editor.scrollTop;
                    var strPos = this.editor.selectionStart;
                    var front = (this.myText).substring(0, strPos);
                    var back = (this.myText).substring(strPos, this.editor.value.length);
                    this.editor.value = front + text + back;
                    strPos = cursorPosition === 0 ? strPos + text.length : strPos + cursorPosition;
                    this.editor.focus();
                    this.editor.selectionStart = strPos;
                    this.editor.selectionEnd = strPos;
                    this.editor.scrollTo({
                        top: scrollPos
                    });
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Boolean)
                ], MarkDownItCustomElement.prototype, "showToolBar", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Boolean)
                ], MarkDownItCustomElement.prototype, "showPreview", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Boolean)
                ], MarkDownItCustomElement.prototype, "showEditor", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], MarkDownItCustomElement.prototype, "src", void 0);
                MarkDownItCustomElement = __decorate([
                    aurelia_framework_1.inject(aurelia_http_client_1.HttpClient),
                    aurelia_framework_1.customElement('aut-mark-down'),
                    __metadata("design:paramtypes", [aurelia_http_client_1.HttpClient])
                ], MarkDownItCustomElement);
                return MarkDownItCustomElement;
            }());
            exports_1("MarkDownItCustomElement", MarkDownItCustomElement);
        }
    };
});
