// aurelia-toolbelt/dist/amd/components/bootstrap/navs/abt-nav-item.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework", "../../../utilities/vanilla/uuid"], function (require, exports, aurelia_framework_1, uuid_1) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var BootstrapNavItem = (function () {
        function BootstrapNavItem(element, uuid) {
            this.uuid = uuid;
            this.active = false;
            this.disabled = false;
            this.isFade = false;
            this.element = element;
        }
        BootstrapNavItem.prototype.attached = function () {
            var navComponent;
            var navs;
            var tab_content;
            navs = this.tab_header.parentElement;
            tab_content = navs.parentElement.parentElement.children.item(1).children.item(0);
            var onlyActiveAttribute = (this.active === '' && this.element.hasAttribute('active'));
            this.active = onlyActiveAttribute || this.active === 'true' || this.active === true;
            var onlyDisabledAttribute = (this.disabled === '' && this.element.hasAttribute('disabled'));
            this.disabled = onlyDisabledAttribute || this.disabled === 'true' || this.disabled === true;
            navs.removeChild(this.tab_body);
            var id = this.element.hasAttribute('id') ? this.element.getAttribute('id') : this.uuid.Uuidv4ForId();
            if (this.tab_body.textContent.length > 8) {
                navComponent = navs.parentElement.parentElement.parentElement;
                var tab_body_id = id + "-tab-body";
                this.isFade =
                    navComponent.hasAttribute('fade') &&
                        (navComponent.getAttribute('fade') === ''
                            ||
                                navComponent.getAttribute('fade') === 'true');
                var isPill = navComponent.hasAttribute('pills') &&
                    (navComponent.getAttribute('pills') === ''
                        ||
                            navComponent.getAttribute('pills') === 'true');
                var isTabs = navComponent.hasAttribute('tabs') &&
                    (navComponent.getAttribute('tabs') === ''
                        ||
                            navComponent.getAttribute('tabs') === 'true');
                var data_toggle = isTabs
                    ? 'tab'
                    : isPill
                        ? 'pill'
                        : '';
                this.tab_header.setAttribute('data-toggle', data_toggle);
                this.tab_header.setAttribute('role', 'tab');
                this.tab_header.setAttribute('aria-controls', "" + tab_body_id);
                this.tab_header.setAttribute('aria-selected', "" + this.active);
                this.tab_header.setAttribute('href', "#" + tab_body_id);
                this.tab_body.setAttribute('id', "" + tab_body_id);
                this.tab_body.setAttribute('aria-labelledby', "" + id);
                tab_content.appendChild(this.tab_body);
            }
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapNavItem.prototype, "class", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapNavItem.prototype, "style", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapNavItem.prototype, "href", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapNavItem.prototype, "title", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapNavItem.prototype, "active", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapNavItem.prototype, "disabled", void 0);
        BootstrapNavItem = __decorate([
            aurelia_framework_1.inject(Element, uuid_1.Uuid),
            aurelia_framework_1.containerless(),
            aurelia_framework_1.customElement('abt-nav-item'),
            __metadata("design:paramtypes", [Element, uuid_1.Uuid])
        ], BootstrapNavItem);
        return BootstrapNavItem;
    }());
    exports.BootstrapNavItem = BootstrapNavItem;
});
