// aurelia-toolbelt/dist/amd/components/bootstrap/collapse/abt-accordion-item.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework"], function (require, exports, aurelia_framework_1) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var BootstrapAccordionItem = (function () {
        function BootstrapAccordionItem(element) {
            this.element = element;
            this.class = '';
            this.style = '';
            this.title = '';
            this.active = false;
        }
        BootstrapAccordionItem.prototype.onAnchorClick = function (event) {
            event.preventDefault();
            return false;
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapAccordionItem.prototype, "class", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapAccordionItem.prototype, "style", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapAccordionItem.prototype, "title", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Boolean)
        ], BootstrapAccordionItem.prototype, "active", void 0);
        BootstrapAccordionItem = __decorate([
            aurelia_framework_1.inject(Element),
            aurelia_framework_1.containerless(),
            aurelia_framework_1.customElement('abt-accordion-item'),
            __metadata("design:paramtypes", [Element])
        ], BootstrapAccordionItem);
        return BootstrapAccordionItem;
    }());
    exports.BootstrapAccordionItem = BootstrapAccordionItem;
});
