// aurelia-toolbelt/dist/system/components/bootstrap/dropdown/abt-dropdown-item.js
System.register(["aurelia-framework", "aurelia-event-aggregator", "./abt-dropdown-selected-item-changed"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_event_aggregator_1, abt_dropdown_selected_item_changed_1, BootstrapDropdownItem;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_event_aggregator_1_1) {
                aurelia_event_aggregator_1 = aurelia_event_aggregator_1_1;
            },
            function (abt_dropdown_selected_item_changed_1_1) {
                abt_dropdown_selected_item_changed_1 = abt_dropdown_selected_item_changed_1_1;
            }
        ],
        execute: function () {
            BootstrapDropdownItem = (function () {
                function BootstrapDropdownItem(ea, element) {
                    this.ea = ea;
                    this.element = element;
                    this.class = '';
                    this.style = '';
                    this.disabled = null;
                }
                BootstrapDropdownItem.prototype.attached = function () {
                    this.dropdownId = this.item.parentElement.parentElement.getAttribute('data-id');
                    this.disabled = this.disabled === '' || this.disabled;
                    if (this.model !== undefined || this.value !== undefined) {
                        var selectedValue = this.model !== undefined
                            ? this.model
                            : this.value !== undefined
                                ? this.value
                                : undefined;
                        this.ea.publish(new abt_dropdown_selected_item_changed_1.BootstrapDropdownSelectedItemChanged(this.dropdownId, selectedValue, this.item.innerText, false));
                    }
                };
                BootstrapDropdownItem.prototype.onClick = function () {
                    if (this.disabled) {
                        return;
                    }
                    if (this.model !== undefined || this.value !== undefined) {
                        var selectedValue = this.model !== undefined
                            ? this.model
                            : this.value !== undefined
                                ? this.value
                                : undefined;
                        this.ea.publish(new abt_dropdown_selected_item_changed_1.BootstrapDropdownSelectedItemChanged(this.dropdownId, selectedValue, this.item.innerText));
                    }
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapDropdownItem.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapDropdownItem.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapDropdownItem.prototype, "disabled", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], BootstrapDropdownItem.prototype, "value", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], BootstrapDropdownItem.prototype, "model", void 0);
                BootstrapDropdownItem = __decorate([
                    aurelia_framework_1.inject(aurelia_event_aggregator_1.EventAggregator, Element),
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('abt-dropdown-item'),
                    __metadata("design:paramtypes", [aurelia_event_aggregator_1.EventAggregator, Element])
                ], BootstrapDropdownItem);
                return BootstrapDropdownItem;
            }());
            exports_1("BootstrapDropdownItem", BootstrapDropdownItem);
        }
    };
});
