// aurelia-toolbelt/dist/es2015/components/vanilla/pretty/aut-radio.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var PrettyRadioButtonCustomElement = (function () {
    function PrettyRadioButtonCustomElement(element, bindingEngine) {
        this.element = element;
        this.bindingEngine = bindingEngine;
        this.disabled = false;
        this.name = '';
        this.switch = false;
        this.outlined = false;
        this.offColor = '';
        this.offLabel = '';
        this.animation = '';
    }
    PrettyRadioButtonCustomElement.prototype.changed = function () {
        if (this.disabled) {
            return;
        }
        this.synchronizeModel();
    };
    PrettyRadioButtonCustomElement.prototype.synchronizeModel = function () {
        this.checked = (this.model !== undefined)
            ? this.model
            : this.value;
    };
    PrettyRadioButtonCustomElement.prototype.checkedChanged = function () {
        this.synchronizeView();
    };
    PrettyRadioButtonCustomElement.prototype.synchronizeView = function () {
        if (this.model !== undefined) {
            this.radioButton.checked = this.matcher
                ? this.matcher(this.checked, this.model)
                : this.checked === this.model;
        }
        else if (this.value) {
            this.radioButton.checked = this.matcher
                ? this.matcher(this.checked, this.value)
                : this.checked === this.value;
        }
    };
    PrettyRadioButtonCustomElement.prototype.bind = function () {
        if (!this.element.hasAttribute) {
            console.warn(this.element);
        }
        this.switch = this.switch === true || this.switch === 'true' || this.element.hasAttribute('switch');
        this.outlined = this.outlined === true || this.outlined === 'true' || this.element.hasAttribute('outline');
        this.disabled = this.disabled === true || this.disabled === 'true' || this.disabled === 'disabled';
        this.colorCss = "p-" + this.color;
        this.offColorCss = this.offColor !== '' ? "p-" + this.offColor : '';
        if (this.switch) {
            if (this.element.hasAttribute('slim')) {
                this.appearanceCSS = 'p-slim';
            }
            else if (this.element.hasAttribute('outline')) {
                this.appearanceCSS = 'p-outline';
            }
            else {
                this.appearanceCSS = 'p-fill';
            }
        }
        else {
            if (this.element.hasAttribute('curve')) {
                this.appearanceCSS = 'p-curve';
            }
            else {
                this.appearanceCSS = 'p-round';
            }
        }
        this.thickCss = this.element.hasAttribute('thick') ? 'p-thick' : '';
        this.thickCss = this.element.hasAttribute('plain') ? 'p-plain' : '';
        this.animationCss = this.animation !== '' ? "p-" + this.animation : '';
        this.synchronizeView();
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "value", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "model", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "checked", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "matcher", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "disabled", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], PrettyRadioButtonCustomElement.prototype, "name", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "switch", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], PrettyRadioButtonCustomElement.prototype, "outlined", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], PrettyRadioButtonCustomElement.prototype, "color", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], PrettyRadioButtonCustomElement.prototype, "offColor", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], PrettyRadioButtonCustomElement.prototype, "offLabel", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], PrettyRadioButtonCustomElement.prototype, "animation", void 0);
    PrettyRadioButtonCustomElement = __decorate([
        aurelia_framework_1.inject(Element, aurelia_framework_1.BindingEngine),
        aurelia_framework_1.customElement('aut-radio'),
        __metadata("design:paramtypes", [Element, aurelia_framework_1.BindingEngine])
    ], PrettyRadioButtonCustomElement);
    return PrettyRadioButtonCustomElement;
}());
exports.PrettyRadioButtonCustomElement = PrettyRadioButtonCustomElement;
