// aurelia-toolbelt/dist/commonjs/components/jquery/metis-menu/aut-metis-menu.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var $ = require("jquery");
require("metismenu");
var JQueryMetisMenu = (function () {
    function JQueryMetisMenu(element) {
        this.element = element;
        this.class = '';
        this.style = '';
    }
    JQueryMetisMenu.prototype.attached = function () {
        var _this = this;
        $(this.metismenu).metisMenu()
            .on('show.metismenu', function (event) {
            console.log("show menu: " + JSON.stringify(event));
            console.log(_this.showMenu);
            var localEvent = _this.showMenu;
            if (localEvent !== null || localEvent !== undefined) {
                Promise.resolve(function () {
                    localEvent(event);
                });
            }
        })
            .on('shown.metismenu', function (event) {
            console.log("shown menu: " + JSON.stringify(event));
            var localEvent = _this.shownMenu;
            if (localEvent !== null || localEvent !== undefined) {
                Promise.resolve(function () {
                    localEvent(event);
                });
            }
        })
            .on('hide.metismenu', function (event) {
            console.log("hide menu: " + JSON.stringify(event));
            var localEvent = _this.hideMenu;
            if (localEvent !== null || localEvent !== undefined) {
                Promise.resolve(function () {
                    localEvent(event);
                });
            }
        })
            .on('hidden.metismenu', function (event) {
            console.log("menu hidden: " + JSON.stringify(event));
            var localEvent = _this.hiddenMenu;
            if (localEvent !== null || localEvent !== undefined) {
                Promise.resolve(function () {
                    localEvent(event);
                });
            }
        });
    };
    JQueryMetisMenu.prototype.detached = function () {
        $(this.metismenu).metisMenu('dispose');
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], JQueryMetisMenu.prototype, "class", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], JQueryMetisMenu.prototype, "style", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], JQueryMetisMenu.prototype, "showMenu", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], JQueryMetisMenu.prototype, "shownMenu", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], JQueryMetisMenu.prototype, "hideMenu", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], JQueryMetisMenu.prototype, "hiddenMenu", void 0);
    JQueryMetisMenu = __decorate([
        aurelia_framework_1.inject(Element),
        aurelia_framework_1.containerless(),
        aurelia_framework_1.customElement('aut-metis-menu'),
        __metadata("design:paramtypes", [Element])
    ], JQueryMetisMenu);
    return JQueryMetisMenu;
}());
exports.JQueryMetisMenu = JQueryMetisMenu;
