// aurelia-toolbelt/dist/system/custom-attributes/vanilla/uuid/aut-uuid.js
System.register(["aurelia-framework", "../../../utilities/vanilla/uuid"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, uuid_1, UuidCustomAttribute;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (uuid_1_1) {
                uuid_1 = uuid_1_1;
            }
        ],
        execute: function () {
            UuidCustomAttribute = (function () {
                function UuidCustomAttribute(element, idgeneratorV4) {
                    this.element = element;
                    this.idgeneratorV4 = idgeneratorV4;
                }
                UuidCustomAttribute.prototype.bind = function () {
                    this.id = 'a' + this.idgeneratorV4.uuidv4().replace(new RegExp('-', 'g'), '');
                    if (this.value) {
                        this.element.setAttribute(this.value, this.id);
                    }
                    else {
                        this.element.id = this.id;
                    }
                };
                UuidCustomAttribute = __decorate([
                    aurelia_framework_1.inject(Element, uuid_1.Uuid),
                    aurelia_framework_1.customAttribute('aut-uuid'),
                    __metadata("design:paramtypes", [Element, uuid_1.Uuid])
                ], UuidCustomAttribute);
                return UuidCustomAttribute;
            }());
            exports_1("UuidCustomAttribute", UuidCustomAttribute);
        }
    };
});
