// aurelia-toolbelt/dist/es2015/components/vanilla/nprogress/aut-nprogress.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var nprogress = require("nprogress");
var aurelia_framework_1 = require("aurelia-framework");
require("nprogress/nprogress.css");
var NProgressLoadingIndicator = (function () {
    function NProgressLoadingIndicator() {
        this.loading = false;
        this.showSpinner = false;
        this.easing = 'ease';
        this.speed = 500;
        this.color = '#753B85';
        this.size = 4;
    }
    NProgressLoadingIndicator.prototype.attached = function () {
        this.showSpinner = this.showSpinner === true || this.showSpinner === 'true';
        nprogress.configure({ showSpinner: this.showSpinner, easing: this.easing, speed: this.speed });
        aurelia_framework_1.DOM.injectStyles("#nprogress .bar {\n      background: " + this.color + " !important;\n      height: " + this.size + "px !important;\n    }\n    #nprogress .peg {\n      box-shadow: 0 0 10px " + this.color + ", 0 0 5px " + this.color + " !important;\n    }\n    #nprogress .spinner-icon {\n      border-top-color: " + this.color + " !important;\n      border-left-color: " + this.color + " !important;\n    }", null, null, 'aut-injected-nprogress');
    };
    NProgressLoadingIndicator.prototype.loadingChanged = function (newValue) {
        if (newValue) {
            nprogress.start();
        }
        else {
            nprogress.done();
        }
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], NProgressLoadingIndicator.prototype, "loading", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], NProgressLoadingIndicator.prototype, "showSpinner", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], NProgressLoadingIndicator.prototype, "easing", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Number)
    ], NProgressLoadingIndicator.prototype, "speed", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], NProgressLoadingIndicator.prototype, "color", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Number)
    ], NProgressLoadingIndicator.prototype, "size", void 0);
    NProgressLoadingIndicator = __decorate([
        aurelia_framework_1.noView(),
        aurelia_framework_1.customElement('aut-nprogress')
    ], NProgressLoadingIndicator);
    return NProgressLoadingIndicator;
}());
exports.NProgressLoadingIndicator = NProgressLoadingIndicator;
