// aurelia-toolbelt/dist/es2015/components/bootstrap/scrollspy/abt-scrollspy.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var $ = require("jquery");
var BootstrapScrollSpy = (function () {
    function BootstrapScrollSpy(element) {
        this.element = element;
        this.offset = 10;
        this.onBody = false;
    }
    BootstrapScrollSpy.prototype.afterAttached = function () {
        var _this = this;
        this.onBody = Boolean(this.onBody);
        this.offset = Number(this.offset);
        if (!this.onBody) {
            $(this.spy).scrollspy({ target: this.target[0] === '#' ? this.target : "#" + this.target, offset: this.offset });
        }
        else {
            $('body').scrollspy({ target: this.target[0] === '#' ? this.target : "#" + this.target, offset: this.offset });
        }
        if (this.bsScrollspy) {
            $(this.spy).on('activate.bs.scrollspy', function () {
                if (_this.bsScrollspy) {
                    _this.bsScrollspy();
                }
            });
        }
    };
    BootstrapScrollSpy.prototype.detached = function () {
        $(this.spy).scrollspy('dispose');
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], BootstrapScrollSpy.prototype, "target", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Number)
    ], BootstrapScrollSpy.prototype, "offset", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapScrollSpy.prototype, "class", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapScrollSpy.prototype, "style", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], BootstrapScrollSpy.prototype, "bsScrollspy", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapScrollSpy.prototype, "onBody", void 0);
    BootstrapScrollSpy = __decorate([
        aurelia_framework_1.inject(Element),
        aurelia_framework_1.containerless(),
        aurelia_framework_1.customElement('abt-scrollspy'),
        __metadata("design:paramtypes", [Element])
    ], BootstrapScrollSpy);
    return BootstrapScrollSpy;
}());
exports.BootstrapScrollSpy = BootstrapScrollSpy;
