// aurelia-toolbelt/dist/system/components/vanilla/raw-html/aut-raw-html.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, RawHtmlRenderer;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            RawHtmlRenderer = (function () {
                function RawHtmlRenderer(element) {
                    this.element = element;
                }
                RawHtmlRenderer.prototype.attached = function () {
                    this.content = this.dummy.innerHTML.replace('<!--slot-->', '');
                    this.dummy.remove();
                };
                RawHtmlRenderer = __decorate([
                    aurelia_framework_1.inject(Element),
                    aurelia_framework_1.customElement('aut-raw-html'),
                    __metadata("design:paramtypes", [Element])
                ], RawHtmlRenderer);
                return RawHtmlRenderer;
            }());
            exports_1("RawHtmlRenderer", RawHtmlRenderer);
        }
    };
});
