// aurelia-toolbelt/dist/system/components/bootstrap/float-input/abt-float-input.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, BootstrapFloatInput;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            BootstrapFloatInput = (function () {
                function BootstrapFloatInput() {
                    this.placeholder = null;
                    this.placeholderFontSize = null;
                    this.labelFontSize = null;
                    this.placeholderOpacity = null;
                    this.placeholderTop = null;
                    this.size = 'md';
                    this.type = 'text';
                    this.labelColor = null;
                    this.placeholderColor = null;
                }
                BootstrapFloatInput.prototype.afterAttached = function () {
                    var id = this.floatInputLabel.id;
                    var fontSize = '';
                    var top = '';
                    if (!this.floatInput.classList.contains('form-control')) {
                        this.floatInput.classList.add('form-control');
                    }
                    if (this.floatInput.classList.contains('form-control-sm')) {
                        this.size = 'sm';
                    }
                    if (this.floatInput.classList.contains('form-control-lg')) {
                        this.size = 'lg';
                    }
                    if (this.size === 'sm') {
                        this.floatInput.classList.add('form-control-sm');
                        fontSize = '90%';
                        top = '.5em';
                    }
                    else if (this.size === 'lg' || this.floatInput.classList.contains('form-control-lg')) {
                        this.floatInput.classList.add('form-control-lg');
                        fontSize = '120%';
                        top = '.7em';
                    }
                    else {
                        this.floatInput.classList.remove('form-control-sm');
                        this.floatInput.classList.remove('form-control-lg');
                        fontSize = '100%';
                        top = '.7em';
                    }
                    if (this.floatInput.classList.contains('form-control')) {
                        this.floatInputLabel.classList.add('has-float-label');
                        var style = "\n                  #" + id + ".has-float-label .form-control:placeholder-shown:not(:focus) + * {\n                    color : " + (this.placeholderColor || 'black') + " !important;\n                    font-size: " + (this.placeholderFontSize || fontSize) + " !important;\n                    opacity: " + (this.placeholderOpacity || '.5') + " !important;\n                    top: " + (this.placeholderTop || top) + " !important;\n                  }\n                  #" + id + ".has-float-label label, #" + id + ".has-float-label > span\n                  {\n                    color : " + (this.labelColor || 'black') + " !important;\n                    font-size: " + (this.labelFontSize || '75%') + " !important;\n                  }";
                        aurelia_framework_1.DOM.injectStyles(style, null, null, 's' + id);
                    }
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "id", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "placeholder", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "placeholderFontSize", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "labelFontSize", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "placeholderOpacity", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "placeholderTop", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "size", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "type", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "labelColor", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapFloatInput.prototype, "placeholderColor", void 0);
                BootstrapFloatInput = __decorate([
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('abt-float-input')
                ], BootstrapFloatInput);
                return BootstrapFloatInput;
            }());
            exports_1("BootstrapFloatInput", BootstrapFloatInput);
        }
    };
});
