// aurelia-toolbelt/dist/system/components/bootstrap/jumbotron/abt-jumbotron.js
System.register(["aurelia-framework", "aurelia-dependency-injection", "aurelia-templating", "aurelia-binding"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_dependency_injection_1, aurelia_templating_1, aurelia_binding_1, BootstrapJumbotron;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_dependency_injection_1_1) {
                aurelia_dependency_injection_1 = aurelia_dependency_injection_1_1;
            },
            function (aurelia_templating_1_1) {
                aurelia_templating_1 = aurelia_templating_1_1;
            },
            function (aurelia_binding_1_1) {
                aurelia_binding_1 = aurelia_binding_1_1;
            }
        ],
        execute: function () {
            BootstrapJumbotron = (function () {
                function BootstrapJumbotron(element) {
                    this.element = element;
                    this.style = '';
                    this.class = '';
                    this.fluid = false;
                }
                BootstrapJumbotron.prototype.attached = function () {
                    var onlyDismissibleAttribute = (this.fluid === '' && this.element.hasAttribute('fluid'));
                    this.fluid = onlyDismissibleAttribute || this.fluid.toString() === 'true';
                };
                __decorate([
                    aurelia_templating_1.bindable({ defaultBindingMode: aurelia_binding_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapJumbotron.prototype, "style", void 0);
                __decorate([
                    aurelia_templating_1.bindable({ defaultBindingMode: aurelia_binding_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapJumbotron.prototype, "class", void 0);
                __decorate([
                    aurelia_templating_1.bindable({ defaultBindingMode: aurelia_binding_1.bindingMode.oneTime }),
                    __metadata("design:type", Object)
                ], BootstrapJumbotron.prototype, "fluid", void 0);
                BootstrapJumbotron = __decorate([
                    aurelia_dependency_injection_1.inject(Element),
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('abt-jumbotron'),
                    __metadata("design:paramtypes", [Element])
                ], BootstrapJumbotron);
                return BootstrapJumbotron;
            }());
            exports_1("BootstrapJumbotron", BootstrapJumbotron);
        }
    };
});
