// aurelia-toolbelt/dist/system/components/bootstrap/badge/abt-badge.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, BootstrapBadge;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            BootstrapBadge = (function () {
                function BootstrapBadge(element) {
                    this.element = element;
                    this.class = '';
                    this.type = 'primary';
                    this.isPill = false;
                }
                BootstrapBadge.prototype.bind = function () {
                    var onlyIsPillAttribute = (this.isPill === '' && this.element.hasAttribute('is-pill'));
                    this.isPill = onlyIsPillAttribute || this.isPill === 'true' || this.isPill === true;
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapBadge.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapBadge.prototype, "type", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapBadge.prototype, "isPill", void 0);
                BootstrapBadge = __decorate([
                    aurelia_framework_1.inject(Element),
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('abt-badge'),
                    __metadata("design:paramtypes", [Element])
                ], BootstrapBadge);
                return BootstrapBadge;
            }());
            exports_1("BootstrapBadge", BootstrapBadge);
        }
    };
});
