// aurelia-toolbelt/dist/amd/components/jquery/metis-menu/aut-metis-menu-group.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework"], function (require, exports, aurelia_framework_1) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var JQueryMetisGroup = (function () {
        function JQueryMetisGroup(element) {
            this.element = element;
            this.groupClass = '';
            this.iconClass = '';
            this.arrowClass = ' fa arrow';
            this.active = false;
            this.showArrow = true;
        }
        JQueryMetisGroup.prototype.attached = function () {
            this.active = this.active === true
                || this.active === 'true'
                || ((this.active === undefined || this.active == null) && this.element.hasAttribute('active'));
            this.showArrow = this.showArrow === true
                || this.showArrow === 'true'
                || ((this.showArrow === undefined || this.showArrow == null) && this.element.hasAttribute('showArrow'));
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryMetisGroup.prototype, "text", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], JQueryMetisGroup.prototype, "groupClass", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], JQueryMetisGroup.prototype, "iconClass", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], JQueryMetisGroup.prototype, "arrowClass", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], JQueryMetisGroup.prototype, "active", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], JQueryMetisGroup.prototype, "showArrow", void 0);
        JQueryMetisGroup = __decorate([
            aurelia_framework_1.inject(Element),
            aurelia_framework_1.containerless(),
            aurelia_framework_1.customElement('aut-metis-menu-group'),
            __metadata("design:paramtypes", [Element])
        ], JQueryMetisGroup);
        return JQueryMetisGroup;
    }());
    exports.JQueryMetisGroup = JQueryMetisGroup;
});
