// aurelia-toolbelt/dist/system/components/bootstrap/modal/abt-modal-header.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, BootstrapModalHeader;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            BootstrapModalHeader = (function () {
                function BootstrapModalHeader() {
                    this.dismissible = false;
                    this.style = '';
                    this.class = '';
                }
                BootstrapModalHeader.prototype.afterAttached = function () {
                    var x = this.header.parentElement.parentElement.parentElement.getAttribute('data-abt-dismissible');
                    this.dismissible = x === 'true';
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapModalHeader.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapModalHeader.prototype, "class", void 0);
                BootstrapModalHeader = __decorate([
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('abt-modal-header')
                ], BootstrapModalHeader);
                return BootstrapModalHeader;
            }());
            exports_1("BootstrapModalHeader", BootstrapModalHeader);
        }
    };
});
