// aurelia-toolbelt/dist/amd/components/bootstrap/dropdown/abt-dropdown.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework", "aurelia-event-aggregator", "./abt-dropdown-selected-item-changed", "jquery", "../../../utilities/vanilla/uuid"], function (require, exports, aurelia_framework_1, aurelia_event_aggregator_1, abt_dropdown_selected_item_changed_1, $, uuid_1) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var BootstrapDropDown = (function () {
        function BootstrapDropDown(element, ea, uuid) {
            this.element = element;
            this.ea = ea;
            this.alignRight = false;
            this.boundary = 'scrollParent';
            this.type = 'primary';
            this.offset = 0;
            this.flip = true;
            this.size = 'md';
            this.placement = '';
            this.split = false;
            this.style = '';
            this.class = '';
            this.disabled = false;
            this.menuClass = '';
            this.menuStyle = '';
            this.title = '';
            this.isBusy = false;
            this.placementClass = '';
            this.itemsValuesOrModels = [];
            this.task = null;
            this.subscription = null;
            this.id = uuid.Uuidv4ForId();
        }
        BootstrapDropDown.prototype.onClicked = function (event) {
            var _this = this;
            event.preventDefault();
            if (!this.click || this.disabled) {
                return;
            }
            if (this.task) {
                return;
            }
            this.isBusy = true;
            this.task = Promise.resolve(this.click({ event: event, target: this.element }))
                .then(function () { return _this.clickCompleted(); }, function () { return _this.clickCompleted(); });
        };
        BootstrapDropDown.prototype.clickCompleted = function () {
            this.task = null;
            this.isBusy = false;
        };
        BootstrapDropDown.prototype.disposeSubscription = function () {
            if (this.subscription !== null) {
                this.subscription.dispose();
                this.subscription = null;
            }
        };
        BootstrapDropDown.prototype.attached = function () {
            this.split = (this.split === '' && this.element.hasAttribute('split')) || this.split.toString() === 'true';
            this.alignRight = (this.alignRight === '' && this.element.hasAttribute('align-right')) || this.alignRight.toString() === 'true';
            this.element.children.item(0).setAttribute('data-id', this.id);
            switch (this.placement) {
                case 'top':
                    this.placementClass = 'dropup';
                    break;
                case 'right':
                    this.placementClass = 'dropright';
                    break;
                case 'left':
                    this.placementClass = 'dropleft';
                    break;
                default:
                    this.placementClass = '';
                    break;
            }
        };
        BootstrapDropDown.prototype.bind = function () {
            var _this = this;
            this.ea.subscribe(abt_dropdown_selected_item_changed_1.BootstrapDropdownSelectedItemChanged, function (changed) {
                if (changed.parentId !== _this.id) {
                    return;
                }
                if (!changed.isValueChanged) {
                    _this.itemsValuesOrModels.push({ value: changed.selectedItem, text: changed.selectedText });
                    return;
                }
                _this.value = changed.selectedItem;
            });
        };
        BootstrapDropDown.prototype.afterAttached = function () {
            var _this = this;
            $(this.dropdown).dropdown();
            if (this.bsShow) {
                $(this.dropdown).on('show.bs.dropdown', function () {
                    if (_this.bsShow) {
                        _this.bsShow();
                    }
                });
            }
            if (this.bsShown) {
                $(this.dropdown).on('shown.bs.dropdown', function () {
                    if (_this.bsShown) {
                        _this.bsShown();
                    }
                });
            }
            if (this.bsHide) {
                $(this.dropdown).on('hide.bs.dropdown', function () {
                    if (_this.bsHide) {
                        _this.bsHide();
                    }
                });
            }
            if (this.bsHidden) {
                $(this.dropdown).on('hidden.bs.dropdown', function () {
                    if (_this.bsHidden) {
                        _this.bsHidden();
                    }
                });
            }
            if (this.value !== undefined) {
                this.valueChanged(this.value);
            }
        };
        BootstrapDropDown.prototype.valueChanged = function (newValue) {
            var _this = this;
            var hasMatcher = (this.matcher !== undefined && this.matcher !== null);
            var found = hasMatcher
                ? this.itemsValuesOrModels.find(function (x) {
                    if (x.value === null || newValue === null) {
                        return x.value === newValue;
                    }
                    return _this.matcher(x.value, newValue);
                })
                : this.itemsValuesOrModels.find(function (x) { return x.value === newValue; });
            if (!found) {
                return;
            }
            this.title = found.text;
            if (this.changed) {
                this.changed({ selected: newValue });
            }
        };
        BootstrapDropDown.prototype.detached = function () {
            this.task = null;
            $(this.dropdown).off('show.bs.tab');
            $(this.dropdown).off('shown.bs.tab');
            $(this.dropdown).off('hide.bs.tab');
            $(this.dropdown).off('hidden.bs.tab');
            $(this.dropdown).dropdown('dispose');
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "alignRight", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "boundary", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "type", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "offset", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Boolean)
        ], BootstrapDropDown.prototype, "flip", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "size", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "placement", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "split", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "style", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "class", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "disabled", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "menuClass", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "menuStyle", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapDropDown.prototype, "title", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "value", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Object)
        ], BootstrapDropDown.prototype, "matcher", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapDropDown.prototype, "click", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapDropDown.prototype, "changed", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapDropDown.prototype, "bsShow", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapDropDown.prototype, "bsShown", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapDropDown.prototype, "bsHide", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapDropDown.prototype, "bsHidden", void 0);
        BootstrapDropDown = __decorate([
            aurelia_framework_1.inject(Element, aurelia_event_aggregator_1.EventAggregator, uuid_1.Uuid),
            aurelia_framework_1.customElement('abt-dropdown'),
            __metadata("design:paramtypes", [Element, aurelia_event_aggregator_1.EventAggregator, uuid_1.Uuid])
        ], BootstrapDropDown);
        return BootstrapDropDown;
    }());
    exports.BootstrapDropDown = BootstrapDropDown;
});
