// aurelia-toolbelt/dist/amd/components/bootstrap/modal/abt-modal.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework", "jquery"], function (require, exports, aurelia_framework_1, $) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var BootstrapModal = (function () {
        function BootstrapModal(element) {
            this.element = element;
            this.class = '';
            this.style = '';
            this.size = 'md';
            this.animate = true;
            this.dismissible = true;
            this.centered = false;
            this.backdrop = true;
            this.keyboard = true;
            this.focus = true;
            this.visible = false;
        }
        BootstrapModal.prototype.setOpenerProperties = function (open) {
            var isString = typeof open === 'string';
            var opener;
            if (isString) {
                opener = document.getElementById(open);
            }
            else {
                opener = open;
            }
            var id = this.modal.id;
            opener.setAttribute('data-toggle', 'modal');
            opener.setAttribute('data-target', "#" + id);
        };
        BootstrapModal.prototype.setEvents = function () {
            var _this = this;
            if (this.bsShow) {
                $(this.modal).on('show.bs.modal', function (e) {
                    if (_this.bsShow) {
                        _this.bsShow({ relatedTarget: e.relatedTarget });
                    }
                });
            }
            if (this.bsShown) {
                $(this.modal).on('shown.bs.modal', function (e) {
                    if (_this.bsShown) {
                        _this.bsShown({ relatedTarget: e.relatedTarget });
                    }
                });
            }
            if (this.bsHide) {
                $(this.modal).on('hide.bs.modal', function () {
                    if (_this.bsHide) {
                        _this.bsHide();
                    }
                });
            }
            if (this.bsHidden) {
                $(this.modal).on('hidden.bs.modal', function () {
                    if (_this.bsHidden) {
                        _this.bsHidden();
                    }
                });
            }
        };
        BootstrapModal.prototype.visibleChanged = function (newValue) {
            var nv = Boolean(newValue);
            if (nv) {
                $(this.modal).modal('show');
                return;
            }
            $(this.modal).modal('hide');
        };
        BootstrapModal.prototype.afterAttached = function () {
            var _this = this;
            this.animate = this.animate === true || this.animate === 'true';
            this.centered = this.centered === true || this.centered === 'true';
            this.dismissible = this.dismissible === true || this.dismissible === 'true';
            this.backdrop = this.backdrop === true || this.backdrop === 'true';
            this.keyboard = this.keyboard === true || this.keyboard === 'true';
            this.focus = this.focus === true || this.focus === 'true';
            this.setEvents();
            if (this.openBy) {
                this.setOpenerProperties(this.openBy);
            }
            else if (this.visible !== undefined) {
                this.visibleChanged(this.visible);
                $(this.modal).on('hide.bs.modal', function () {
                    _this.visible = false;
                });
            }
            else {
                throw Error("The 'abt-modal' should have either 'open-by' or 'visible' attribute");
            }
            $(this.modal).modal({
                backdrop: this.backdrop,
                keyboard: this.keyboard,
                focus: this.focus,
                show: false
            });
        };
        BootstrapModal.prototype.detached = function () {
            $(this.modal).off('show.bs.modal');
            $(this.modal).off('shown.bs.modal');
            $(this.modal).off('hide.bs.modal');
            $(this.modal).off('hidden.bs.modal');
            $(this.modal).modal('hide');
            $(this.modal).modal('dispose');
            $(this.modal).remove();
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapModal.prototype, "class", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapModal.prototype, "style", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapModal.prototype, "size", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "animate", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "dismissible", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "centered", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "backdrop", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "keyboard", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "focus", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", HTMLElement)
        ], BootstrapModal.prototype, "openBy", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Object)
        ], BootstrapModal.prototype, "visible", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapModal.prototype, "bsShow", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapModal.prototype, "bsShown", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapModal.prototype, "bsHide", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapModal.prototype, "bsHidden", void 0);
        BootstrapModal = __decorate([
            aurelia_framework_1.inject(Element),
            aurelia_framework_1.customElement('abt-modal'),
            __metadata("design:paramtypes", [Element])
        ], BootstrapModal);
        return BootstrapModal;
    }());
    exports.BootstrapModal = BootstrapModal;
});
