// aurelia-toolbelt/dist/system/components/bootstrap/tokenize/abt-tokenize.js
System.register(["aurelia-framework", "jquery", "aureliatoolbelt-thirdparty/bootstrap-tokenize2/tokenize2.js", "../../../utilities/vanilla/jsTools"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, jsTools_1, BootstrapTokenizeCustomElement;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (_1) {
            },
            function (_2) {
            },
            function (jsTools_1_1) {
                jsTools_1 = jsTools_1_1;
            }
        ],
        execute: function () {
            BootstrapTokenizeCustomElement = (function () {
                function BootstrapTokenizeCustomElement(element, jsTools, bindingEngine) {
                    this.element = element;
                    this.jsTools = jsTools;
                    this.bindingEngine = bindingEngine;
                    this.id = null;
                    this.debounce = 0;
                    this.delimiter = [','];
                    this.placeholder = null;
                    this.tokensMaxItems = 0;
                    this.tokensAllowCustom = false;
                    this.dropdownMaxItems = 10;
                    this.searchMinLength = 0;
                    this.searchFromStart = true;
                    this.searchHighlight = true;
                    this.showOnClick = false;
                    this.displayNoResultsMessage = false;
                    this.noResultsMessageText = 'No results matched "%s"';
                    this.zIndexMargin = 500;
                    this.tabIndex = 0;
                }
                BootstrapTokenizeCustomElement.prototype.bind = function () {
                    var _this = this;
                    if (this.selectedTokens) {
                        this.subscription = this.bindingEngine.collectionObserver(this.selectedTokens)
                            .subscribe(function (x) {
                            var item = x[0];
                            if (item) {
                                if (item.removed.length) {
                                    var removed = item.removed;
                                    for (var index = 0; index < removed.length; index++) {
                                        $(_this.tokenize).trigger('tokenize:tokens:remove', removed[index].value);
                                    }
                                }
                                else {
                                    $(_this.tokenize).trigger('tokenize:tokens:add', [_this.selectedTokens[item.index].value,
                                        _this.selectedTokens[item.index].text, _this.selectedTokens[item.index].force || true]);
                                }
                            }
                        });
                    }
                };
                BootstrapTokenizeCustomElement.prototype.detached = function () {
                    this.subscription = null;
                };
                BootstrapTokenizeCustomElement.prototype.afterAttached = function () {
                    var _this = this;
                    if (this.id) {
                        this.tokenizeTemplate.setAttribute('id', "abt-tokenize-" + this.id);
                    }
                    this.debounce = Number(this.debounce);
                    this.tokensAllowCustom = (this.tokensAllowCustom === '' && this.tokenizeTemplate.hasAttribute('tokens-allow-custom'))
                        || this.tokensAllowCustom.toString() === 'true';
                    this.dropdownMaxItems = Number(this.dropdownMaxItems) <= 0 ? 0 : (Number(this.dropdownMaxItems) - 1);
                    this.searchMinLength = Number(this.searchMinLength);
                    this.searchFromStart = (this.searchFromStart === '' && this.tokenizeTemplate.hasAttribute('search-from-start'))
                        || this.searchFromStart.toString() === 'true';
                    this.searchHighlight = (this.searchHighlight === '' && this.tokenizeTemplate.hasAttribute('search-highlight'))
                        || this.searchHighlight.toString() === 'true';
                    this.showOnClick = (this.showOnClick === '' && this.tokenizeTemplate.hasAttribute('show-on-click'))
                        || this.showOnClick.toString() === 'true';
                    this.displayNoResultsMessage = (this.displayNoResultsMessage === '' && this.tokenizeTemplate.hasAttribute('display-no-results-message'))
                        || this.displayNoResultsMessage.toString() === 'true';
                    this.zIndexMargin = Number(this.zIndexMargin);
                    this.tabIndex = Number(this.tabIndex);
                    this.tokensMaxItems = Number(this.tokensMaxItems) || 0;
                    $(this.tokenize).on('tokenize:load', function () {
                        if (_this.load) {
                            _this.load();
                        }
                    });
                    $(this.tokenize).on('tokenize:clear', function () {
                        if (_this.clear) {
                            _this.clear();
                        }
                    });
                    $(this.tokenize).on('tokenize:remap', function () {
                        if (_this.remap) {
                            _this.remap();
                        }
                    });
                    $(this.tokenize).on('tokenize:select', function () {
                        if (_this.select) {
                            _this.select();
                        }
                    });
                    $(this.tokenize).on('tokenize:deselect', function () {
                        if (_this.deselect) {
                            _this.deselect();
                        }
                    });
                    $(this.tokenize).on('tokenize:search', function (e, value) {
                        if (_this.search) {
                            _this.search({ e: e, value: value });
                        }
                    });
                    $(this.tokenize).on('tokenize:paste', function () {
                        if (_this.paste) {
                            _this.paste();
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:up', function () {
                        if (_this.dropdownUp) {
                            _this.dropdownUp();
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:down', function () {
                        if (_this.dropdownDown) {
                            _this.dropdownDown();
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:clear', function () {
                        if (_this.dropdownClear) {
                            _this.dropdownClear();
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:show', function () {
                        if (_this.dropdownShow) {
                            _this.dropdownShow();
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:hide', function () {
                        if (_this.dropdownHide) {
                            _this.dropdownHide();
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:fill', function (e, items) {
                        if (_this.dropdownFill) {
                            _this.dropdownFill({ e: e, items: items });
                        }
                    });
                    $(this.tokenize).on('tokenize:dropdown:itemAdd', function (e, item) {
                        if (_this.dropdownItemAdd) {
                            _this.dropdownItemAdd({ e: e, item: item });
                        }
                    });
                    $(this.tokenize).on('tokenize:keypress', function (e, routedEvent) {
                        if (_this.keypress) {
                            _this.keypress({ e: e, routedEvent: routedEvent });
                        }
                    });
                    $(this.tokenize).on('tokenize:keydown', function (e, routedEvent) {
                        if (_this.keydown) {
                            _this.keydown({ e: e, routedEvent: routedEvent });
                        }
                    });
                    $(this.tokenize).on('tokenize:keyup', function (e, routedEvent) {
                        if (_this.keyup) {
                            _this.keyup({ e: e, routedEvent: routedEvent });
                        }
                    });
                    $(this.tokenize).on('tokenize:tokens:reorder', function () {
                        if (_this.reorder) {
                            _this.reorder();
                        }
                    });
                    $(this.tokenize).on('tokenize:tokens:add', function (e, value, text, force) {
                        if (_this.selectedTokens) {
                            var found = _this.selectedTokens.findIndex(function (x) { return x.value === value; }) > -1;
                            if (!found) {
                                _this.selectedTokens.push({ text: text, value: value });
                            }
                        }
                        if (_this.add) {
                            _this.add({ e: e, value: value, text: text, force: force });
                        }
                    });
                    $(this.tokenize).on('tokenize:tokens:remove', function (e, value) {
                        if (_this.selectedTokens) {
                            var index = _this.selectedTokens.findIndex(function (x) { return x.value === value; });
                            if (index > -1) {
                                _this.selectedTokens.splice(index, 1);
                            }
                        }
                        if (_this.remove) {
                            _this.remove({ e: e, value: value });
                        }
                    });
                    var ds = null;
                    if (this.options) {
                        var index = this.options.length;
                        while (index--) {
                            this.options[index].innerHTML = this.options[index].innerHTML.trim();
                            this.tokenize.appendChild(this.options[index]);
                        }
                        ds = 'select';
                    }
                    if (!this.options && this.jsTools.isString(this.dataSource)) {
                        ds = this.dataSource;
                    }
                    if (this.jsTools.isFunction(this.dataSource)) {
                        ds = null;
                    }
                    $(this.tokenize).tokenize2({
                        dataSource: ds != null ? ds : function (term, object) {
                            var items = [];
                            var filterd = _this.dataSource({ term: term });
                            $.each(filterd, function (_k, v) {
                                items.push(v);
                            });
                            var data = [items];
                            object.trigger('tokenize:dropdown:fill', data);
                        },
                        debounce: this.debounce,
                        delimiter: this.delimiter,
                        placeholder: this.placeholder,
                        tokensMaxItems: this.tokensMaxItems,
                        tokensAllowCustom: this.tokensAllowCustom,
                        dropdownMaxItems: this.dropdownMaxItems,
                        searchMinLength: this.searchMinLength,
                        searchFromStart: this.searchFromStart,
                        searchHighlight: this.searchHighlight,
                        displayNoResultsMessage: this.displayNoResultsMessage,
                        noResultsMessageText: this.noResultsMessageText,
                        zIndexMargin: this.zIndexMargin,
                        tabIndex: this.tabIndex
                    });
                    if (this.showOnClick) {
                        $(this.tokenize).on('tokenize:select', function (e, routedEvent) {
                            $(_this.tokenize).trigger('tokenize:search', '');
                        });
                    }
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapTokenizeCustomElement.prototype, "id", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapTokenizeCustomElement.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapTokenizeCustomElement.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "debounce", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Array)
                ], BootstrapTokenizeCustomElement.prototype, "delimiter", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapTokenizeCustomElement.prototype, "placeholder", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "tokensMaxItems", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "tokensAllowCustom", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownMaxItems", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "searchMinLength", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "searchFromStart", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "searchHighlight", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "showOnClick", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "displayNoResultsMessage", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapTokenizeCustomElement.prototype, "noResultsMessageText", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "zIndexMargin", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "tabIndex", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], BootstrapTokenizeCustomElement.prototype, "dataSource", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Array)
                ], BootstrapTokenizeCustomElement.prototype, "selectedTokens", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "load", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "clear", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "remap", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "select", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "deselect", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "search", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "paste", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownUp", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownDown", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownClear", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownShow", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownHide", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownFill", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "dropdownItemAdd", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "keypress", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "keydown", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "keyup", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "reorder", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "add", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapTokenizeCustomElement.prototype, "remove", void 0);
                __decorate([
                    aurelia_framework_1.children('option'),
                    __metadata("design:type", Array)
                ], BootstrapTokenizeCustomElement.prototype, "options", void 0);
                BootstrapTokenizeCustomElement = __decorate([
                    aurelia_framework_1.inject(Element, jsTools_1.JsTools, aurelia_framework_1.BindingEngine),
                    aurelia_framework_1.customElement('abt-tokenize'),
                    __metadata("design:paramtypes", [Element, jsTools_1.JsTools, aurelia_framework_1.BindingEngine])
                ], BootstrapTokenizeCustomElement);
                return BootstrapTokenizeCustomElement;
            }());
            exports_1("BootstrapTokenizeCustomElement", BootstrapTokenizeCustomElement);
        }
    };
});
