// aurelia-toolbelt/dist/es2015/components/bootstrap/star-rate/abt-star-rate.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var BootstrapStarRate = (function () {
    function BootstrapStarRate(element) {
        this.element = element;
        this.rtl = false;
        this.style = '';
        this.class = '';
        this.type = 'primary';
        this.color = null;
        this.maxRate = 5;
        this.disabled = false;
        this.fullStar = 'fa fa-star';
        this.emptyStar = 'fa fa-star-o';
        this.halfStar = false;
        this.rate = 0;
        this.mouseRate = -1;
        this.showHalfStar = false;
        this.halfStarClass = null;
    }
    BootstrapStarRate.prototype.bind = function () {
        var onlyDisabledAttribute = (this.disabled === '' && this.element.hasAttribute('read-only'));
        this.disabled = onlyDisabledAttribute || this.disabled.toString() === 'true';
        var onlyRTLAttribute = (this.rtl === '' && this.element.hasAttribute('rtl'));
        this.rtl = onlyRTLAttribute || this.rtl.toString() === 'true';
        var onlyHalfStarAttribute = (this.halfStar === '' && this.element.hasAttribute('half-star'));
        this.halfStar = onlyHalfStarAttribute || this.halfStar.toString() === 'true';
        this.maxRate = Number(this.maxRate);
        this.rate = Number(this.rate);
    };
    BootstrapStarRate.prototype.halfStarChanged = function (newValue) {
        this.halfStarClass = newValue ? 'fa fa-star-half-o' : null;
    };
    BootstrapStarRate.prototype.mouseMove = function (event, index) {
        if (this.disabled) {
            return;
        }
        if (this.halfStarClass) {
            var calculatedIndex = this.rtl ? this.maxRate - index - 1 : index;
            var controlLeft = this.rtl ? this.icons[calculatedIndex].getBoundingClientRect().right : this.icons[calculatedIndex].getBoundingClientRect().left;
            var currentMousePosition = this.rtl ? controlLeft - event.clientX : event.clientX - controlLeft;
            this.showHalfStar = currentMousePosition < (this.icons[calculatedIndex].clientWidth / 2);
        }
        this.mouseRate = index + 1 - (this.showHalfStar ? 0.5 : 0);
    };
    BootstrapStarRate.prototype.setRate = function (index) {
        if (this.disabled) {
            return;
        }
        var oldValue = this.rate;
        this.rate = index + 1 - (this.showHalfStar ? 0.5 : 0);
        if (oldValue !== this.rate && this.rateChanged) {
            this.rateChanged({ newRate: this.rate, oldRate: oldValue });
        }
    };
    BootstrapStarRate.prototype.mouseLeft = function () {
        if (this.disabled) {
            return;
        }
        this.showHalfStar = false;
        this.mouseRate = -1;
    };
    Object.defineProperty(BootstrapStarRate.prototype, "currentValue", {
        get: function () {
            var x = (this.mouseRate !== -1 ? this.mouseRate : this.rate);
            return x;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BootstrapStarRate.prototype, "hasFloatingPoint", {
        get: function () {
            var mode = this.currentValue % 1;
            return mode > 0 && mode < 1;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BootstrapStarRate.prototype, "fixedPoint", {
        get: function () {
            return Math.floor(this.currentValue);
        },
        enumerable: true,
        configurable: true
    });
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapStarRate.prototype, "rtl", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapStarRate.prototype, "style", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapStarRate.prototype, "class", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapStarRate.prototype, "type", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapStarRate.prototype, "color", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Number)
    ], BootstrapStarRate.prototype, "maxRate", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], BootstrapStarRate.prototype, "disabled", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapStarRate.prototype, "fullStar", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapStarRate.prototype, "emptyStar", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapStarRate.prototype, "halfStar", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Number)
    ], BootstrapStarRate.prototype, "rate", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], BootstrapStarRate.prototype, "rateChanged", void 0);
    __decorate([
        aurelia_framework_1.children('i'),
        __metadata("design:type", Array)
    ], BootstrapStarRate.prototype, "icons", void 0);
    __decorate([
        aurelia_framework_1.computedFrom('mouseRate', 'rate'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], BootstrapStarRate.prototype, "currentValue", null);
    __decorate([
        aurelia_framework_1.computedFrom('currentValue'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], BootstrapStarRate.prototype, "hasFloatingPoint", null);
    __decorate([
        aurelia_framework_1.computedFrom('currentValue'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], BootstrapStarRate.prototype, "fixedPoint", null);
    BootstrapStarRate = __decorate([
        aurelia_framework_1.inject(Element),
        aurelia_framework_1.customElement('abt-star-rate'),
        __metadata("design:paramtypes", [Element])
    ], BootstrapStarRate);
    return BootstrapStarRate;
}());
exports.BootstrapStarRate = BootstrapStarRate;
