// aurelia-toolbelt/dist/es2015/components/bootstrap/navs/abt-navs.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var aurelia_templating_1 = require("aurelia-templating");
var $ = require("jquery");
var BootstrapNavs = (function () {
    function BootstrapNavs(element) {
        this.element = element;
        this.navsVerticalClass = 'col-sm-3';
        this.contentVerticalClass = 'col-sm-9';
        this.tabs = false;
        this.pills = false;
        this.vertical = false;
        this.justify = false;
        this.fill = false;
        this.beTab = true;
        this.bePills = false;
    }
    BootstrapNavs.prototype.afterAttached = function () {
        var onlyPillsAttribute = (this.pills === '' && this.element.hasAttribute('pills'));
        this.pills = onlyPillsAttribute || this.pills.toString() === 'true';
        var onlyTabsAttribute = (this.tabs === '' && this.element.hasAttribute('tabs'));
        this.tabs = onlyTabsAttribute || this.tabs.toString() === 'true';
        var onlyVerticalAttribute = (this.vertical === '' && this.element.hasAttribute('vertical'));
        this.vertical = onlyVerticalAttribute || this.vertical.toString() === 'true';
        var onlyJustifiedAttribute = (this.justify === '' && this.element.hasAttribute('justify'));
        this.justify = onlyJustifiedAttribute || this.justify.toString() === 'true';
        var onlyFillAttribute = (this.fill === '' && this.element.hasAttribute('fill'));
        this.fill = onlyFillAttribute || this.fill.toString() === 'true';
        this.beTab = this.tabs;
        this.bePills = this.pills;
        if (this.justify && this.fill) {
            var error = new Error("The [abt-navs] should have either 'fill' or 'justify' attributes, and not both of them simultaneously.");
            throw error;
        }
        var children = this.element.children.item(0).children.item(0).getElementsByTagName('a');
        $(children).tab();
        this.handle_events();
    };
    BootstrapNavs.prototype.handle_events = function () {
        var _this = this;
        var children = this.element.children.item(0).children.item(0).getElementsByTagName('a');
        if (this.bsShow) {
            $(children).on('show.bs.tab', function (event) {
                if (_this.bsShow) {
                    _this.bsShow({ activeTab: event.target, prevTab: event.relatedTarget });
                }
            });
        }
        if (this.bsShown) {
            $(children).on('shown.bs.tab', function (event) {
                if (_this.bsShown) {
                    _this.bsShown({ activeTab: event.target, prevTab: event.relatedTarget });
                }
            });
        }
        if (this.bsHide) {
            $(children).on('hide.bs.tab', function (event) {
                if (_this.bsHide) {
                    _this.bsHide({ activeTab: event.target, prevTab: event.relatedTarget });
                }
            });
        }
        if (this.bsHidden) {
            $(children).on('hidden.bs.tab', function (event) {
                if (_this.bsHidden) {
                    _this.bsHidden({ activeTab: event.target, prevTab: event.relatedTarget });
                }
            });
        }
    };
    BootstrapNavs.prototype.detached = function () {
        var children = this.element.children.item(0).children.item(0).getElementsByTagName('a');
        $(children).off('show.bs.tab');
        $(children).off('shown.bs.tab');
        $(children).off('hide.bs.tab');
        $(children).off('hidden.bs.tab');
        $(children).tab('dispose');
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapNavs.prototype, "navsVerticalClass", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapNavs.prototype, "contentVerticalClass", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapNavs.prototype, "navsClass", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapNavs.prototype, "navsStyle", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapNavs.prototype, "contentClass", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapNavs.prototype, "contentStyle", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Function)
    ], BootstrapNavs.prototype, "bsShow", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Function)
    ], BootstrapNavs.prototype, "bsHide", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Function)
    ], BootstrapNavs.prototype, "bsShown", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Function)
    ], BootstrapNavs.prototype, "bsHidden", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapNavs.prototype, "tabs", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapNavs.prototype, "pills", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapNavs.prototype, "vertical", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapNavs.prototype, "justify", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapNavs.prototype, "fill", void 0);
    BootstrapNavs = __decorate([
        aurelia_framework_1.inject(Element),
        aurelia_templating_1.customElement('abt-navs'),
        __metadata("design:paramtypes", [Element])
    ], BootstrapNavs);
    return BootstrapNavs;
}());
exports.BootstrapNavs = BootstrapNavs;
