// aurelia-toolbelt/dist/system/components/vanilla/pretty/aut-checkbox.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, PrettyCheckboxCustomElement;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            PrettyCheckboxCustomElement = (function () {
                function PrettyCheckboxCustomElement(element, bindingEngine) {
                    this.element = element;
                    this.bindingEngine = bindingEngine;
                    this.disabled = false;
                    this.name = '';
                    this.switch = false;
                    this.outlined = false;
                    this.offColor = '';
                    this.offLabel = '';
                    this.animation = '';
                    this.subscription = null;
                }
                PrettyCheckboxCustomElement.prototype.changed = function () {
                    if (this.disabled) {
                        return;
                    }
                    this.state = !this.state;
                    this.synchronizeModel(this.state);
                };
                PrettyCheckboxCustomElement.prototype.synchronizeModel = function (newState) {
                    var _this = this;
                    if (!Array.isArray(this.checked)) {
                        this.checked = newState;
                        return;
                    }
                    if (newState && ((this.matcher && this.checked.findIndex(function (x) { return _this.matcher(x, _this.value || _this.model); }) === -1)
                        ||
                            (this.checked.indexOf(this.value || this.model) === -1))) {
                        this.checked.push(this.value || this.model);
                    }
                    else if (!newState) {
                        var index = this.matcher
                            ? this.checked.findIndex(function (x) { return _this.matcher(x, _this.value || _this.model); })
                            : this.checked.indexOf(this.value || this.model);
                        if (index !== -1) {
                            this.checked.splice(index, 1);
                        }
                    }
                };
                PrettyCheckboxCustomElement.prototype.checkedChanged = function (newValue) {
                    var _this = this;
                    this.disposeSubscription();
                    if (Array.isArray(this.checked)) {
                        this.subscription = this.bindingEngine.collectionObserver(this.checked)
                            .subscribe(function () {
                            _this.synchronizeView(newValue);
                        });
                    }
                    this.synchronizeView(newValue);
                };
                PrettyCheckboxCustomElement.prototype.synchronizeView = function (newValue) {
                    var _this = this;
                    if (Array.isArray(this.checked)) {
                        var index = this.matcher
                            ? this.checked.findIndex(function (x) { return _this.matcher(x, _this.value || _this.model); })
                            : this.checked.indexOf(this.value || this.model);
                        this.state = index !== -1;
                    }
                    else {
                        this.state = newValue;
                    }
                };
                PrettyCheckboxCustomElement.prototype.disposeSubscription = function () {
                    if (this.subscription !== null) {
                        this.subscription.dispose();
                        this.subscription = null;
                    }
                };
                PrettyCheckboxCustomElement.prototype.bind = function () {
                    if (!this.element.hasAttribute) {
                        console.warn(this.element);
                    }
                    this.switch = this.switch === true || this.switch === 'true' || this.element.hasAttribute('switch');
                    this.outlined = this.outlined === true || this.outlined === 'true' || this.element.hasAttribute('outlined');
                    this.disabled = this.disabled === true || this.disabled === 'true' || this.disabled === 'disabled';
                    this.colorCss = "p-" + this.color;
                    this.offColorCss = this.offColor !== '' ? "p-" + this.offColor : '';
                    if (this.switch) {
                        if (this.element.hasAttribute('slim')) {
                            this.appearanceCSS = 'p-slim';
                        }
                        else if (this.element.hasAttribute('outlined')) {
                            this.appearanceCSS = 'p-outline';
                        }
                        else {
                            this.appearanceCSS = 'p-fill';
                        }
                    }
                    else {
                        if (this.element.hasAttribute('curve')) {
                            this.appearanceCSS = 'p-curve';
                        }
                        else if (this.element.hasAttribute('round')) {
                            this.appearanceCSS = 'p-round';
                        }
                        else {
                            this.appearanceCSS = '';
                        }
                    }
                    this.thickCss = this.element.hasAttribute('thick') ? 'p-thick' : '';
                    this.thickCss = this.element.hasAttribute('plain') ? 'p-plain' : '';
                    this.animationCss = this.animation !== '' ? "p-" + this.animation : '';
                    this.isCheckBox = !this.element.hasAttribute('radio');
                    this.synchronizeView(this.checked);
                };
                PrettyCheckboxCustomElement.prototype.unbind = function () {
                    this.disposeSubscription();
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "value", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "model", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "checked", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "matcher", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "disabled", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", String)
                ], PrettyCheckboxCustomElement.prototype, "name", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "switch", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", Object)
                ], PrettyCheckboxCustomElement.prototype, "outlined", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], PrettyCheckboxCustomElement.prototype, "color", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], PrettyCheckboxCustomElement.prototype, "offColor", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], PrettyCheckboxCustomElement.prototype, "offLabel", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], PrettyCheckboxCustomElement.prototype, "animation", void 0);
                PrettyCheckboxCustomElement = __decorate([
                    aurelia_framework_1.inject(Element, aurelia_framework_1.BindingEngine),
                    aurelia_framework_1.customElement('aut-checkbox'),
                    __metadata("design:paramtypes", [Element, aurelia_framework_1.BindingEngine])
                ], PrettyCheckboxCustomElement);
                return PrettyCheckboxCustomElement;
            }());
            exports_1("PrettyCheckboxCustomElement", PrettyCheckboxCustomElement);
        }
    };
});
