// aurelia-toolbelt/dist/commonjs/custom-attributes/vanilla/id/aut-id.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var sharedIndex_1 = require("./../../../utilities/vanilla/sharedIndex");
var aurelia_framework_1 = require("aurelia-framework");
var uuid_1 = require("../../../utilities/vanilla/uuid");
var IdCustomAttribute = (function () {
    function IdCustomAttribute(element, sharedIndex, idgeneratorV4) {
        this.element = element;
        this.sharedIndex = sharedIndex;
        this.idgeneratorV4 = idgeneratorV4;
    }
    IdCustomAttribute.prototype.bind = function () {
        var uuid = this.idgeneratorV4.Uuidv4ForId();
        var group = this.element.getAttribute('data-aut-id-group');
        if (!group) {
            this.element.setAttribute(uuid, 'data-aut-id-group');
            group = uuid;
        }
        this.id = 'a' + this.sharedIndex.getAndIncrement(group);
        this.element.id = this.id;
    };
    IdCustomAttribute = __decorate([
        aurelia_framework_1.inject(Element, sharedIndex_1.SharedIndex, uuid_1.Uuid),
        aurelia_framework_1.customAttribute('aut-id'),
        __metadata("design:paramtypes", [Element, sharedIndex_1.SharedIndex, uuid_1.Uuid])
    ], IdCustomAttribute);
    return IdCustomAttribute;
}());
exports.IdCustomAttribute = IdCustomAttribute;
