// aurelia-toolbelt/dist/es2015/components/bootstrap/button/abt-link-button.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var aurelia_framework_1 = require("aurelia-framework");
var BootstrapLinkButton = (function () {
    function BootstrapLinkButton(element) {
        this.element = element;
        this.type = 'primary';
        this.size = 'md';
        this.outline = false;
        this.block = false;
        this.style = '';
        this.class = '';
        this.href = '';
        this.isBusy = false;
        this.task = null;
    }
    BootstrapLinkButton.prototype.afterAttached = function () {
        var onlyOutlineAttribute = (this.outline === '' && this.element.hasAttribute('outline'));
        this.outline = onlyOutlineAttribute || this.outline === 'true' || this.outline === true;
        var onlyBlockAttribute = (this.block === '' && this.element.hasAttribute('block'));
        this.block = onlyBlockAttribute || this.block === 'true' || this.block === true;
        var onlyDisabledAttribute = (this.disabled === '' && this.element.hasAttribute('disabled'));
        this.disabled = onlyDisabledAttribute || this.disabled === 'true' || this.disabled === true;
    };
    BootstrapLinkButton.prototype.onClick = function (event) {
        var _this = this;
        if (!this.click) {
            return true;
        }
        event.preventDefault();
        if (this.task || this.disabled) {
            return false;
        }
        this.isBusy = true;
        this.task = Promise.resolve(this.click({ event: event, target: event.target }))
            .then(function () { return _this.clickCompleted(); }, function () { return _this.clickCompleted(); });
        return false;
    };
    BootstrapLinkButton.prototype.clickCompleted = function () {
        this.task = null;
        this.isBusy = false;
    };
    BootstrapLinkButton.prototype.detached = function () {
        this.task = null;
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], BootstrapLinkButton.prototype, "type", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], BootstrapLinkButton.prototype, "size", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapLinkButton.prototype, "outline", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", Object)
    ], BootstrapLinkButton.prototype, "block", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapLinkButton.prototype, "style", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapLinkButton.prototype, "class", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], BootstrapLinkButton.prototype, "href", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], BootstrapLinkButton.prototype, "id", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Function)
    ], BootstrapLinkButton.prototype, "click", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
        __metadata("design:type", Object)
    ], BootstrapLinkButton.prototype, "disabled", void 0);
    BootstrapLinkButton = __decorate([
        aurelia_framework_1.inject(Element),
        aurelia_framework_1.containerless(),
        aurelia_framework_1.customElement('abt-link-button'),
        __metadata("design:paramtypes", [Element])
    ], BootstrapLinkButton);
    return BootstrapLinkButton;
}());
exports.BootstrapLinkButton = BootstrapLinkButton;
