// aurelia-toolbelt/dist/system/components/bootstrap/navbar/abt-navbar-brand.js
System.register(["aurelia-framework", "aurelia-templating"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_templating_1, BootstrapNavBarBrand;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_templating_1_1) {
                aurelia_templating_1 = aurelia_templating_1_1;
            }
        ],
        execute: function () {
            BootstrapNavBarBrand = (function () {
                function BootstrapNavBarBrand() {
                    this.heading = false;
                }
                BootstrapNavBarBrand.prototype.onClick = function (event) {
                    if (this.click) {
                        this.click({ event: event });
                    }
                    return true;
                };
                BootstrapNavBarBrand.prototype.afterAttached = function () {
                    var isHeading = Boolean(this.heading) || this.navbarBrandTemplate.hasAttribute('heading');
                    if (isHeading) {
                        this.navbarBrand.classList.add('abt-navbar-brand-heading');
                    }
                    else {
                        this.navbarBrand.classList.remove('abt-navbar-brand-heading');
                    }
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarBrand.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarBrand.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapNavBarBrand.prototype, "click", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapNavBarBrand.prototype, "id", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarBrand.prototype, "href", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapNavBarBrand.prototype, "heading", void 0);
                BootstrapNavBarBrand = __decorate([
                    aurelia_framework_1.inject(Element),
                    aurelia_framework_1.containerless(),
                    aurelia_templating_1.customElement('abt-navbar-brand')
                ], BootstrapNavBarBrand);
                return BootstrapNavBarBrand;
            }());
            exports_1("BootstrapNavBarBrand", BootstrapNavBarBrand);
        }
    };
});
