// aurelia-toolbelt/dist/system/components/bootstrap/navbar/abt-navbar-link.js
System.register(["aurelia-framework", "aurelia-templating"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_templating_1, BootstrapNavBarLink;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_templating_1_1) {
                aurelia_templating_1 = aurelia_templating_1_1;
            }
        ],
        execute: function () {
            BootstrapNavBarLink = (function () {
                function BootstrapNavBarLink() {
                    this.active = false;
                    this.disabled = false;
                }
                BootstrapNavBarLink.prototype.afterAttached = function () {
                    var isActive = Boolean(this.active) || this.navItem.hasAttribute('active');
                    var isDisabled = Boolean(this.disabled) || this.navItem.hasAttribute('disabled');
                    if (isActive) {
                        this.navItemLink.classList.add('active');
                        this.navItemLink.classList.remove('disabled');
                    }
                    if (isDisabled) {
                        this.navItemLink.classList.remove('active');
                        this.navItemLink.classList.add('disabled');
                    }
                };
                BootstrapNavBarLink.prototype.onClick = function (event) {
                    if (this.click) {
                        this.click({ event: event });
                    }
                    return true;
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarLink.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarLink.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapNavBarLink.prototype, "id", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarLink.prototype, "href", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarLink.prototype, "linkClass", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapNavBarLink.prototype, "linkStyle", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapNavBarLink.prototype, "active", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapNavBarLink.prototype, "disabled", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapNavBarLink.prototype, "click", void 0);
                BootstrapNavBarLink = __decorate([
                    aurelia_framework_1.inject(Element),
                    aurelia_framework_1.containerless(),
                    aurelia_templating_1.customElement('abt-navbar-link')
                ], BootstrapNavBarLink);
                return BootstrapNavBarLink;
            }());
            exports_1("BootstrapNavBarLink", BootstrapNavBarLink);
        }
    };
});
