// aurelia-toolbelt/dist/system/components/jquery/block-ui/sharedOptions.js
System.register(["aurelia-dependency-injection"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_dependency_injection_1, SharedOptions;
    return {
        setters: [
            function (aurelia_dependency_injection_1_1) {
                aurelia_dependency_injection_1 = aurelia_dependency_injection_1_1;
            }
        ],
        execute: function () {
            SharedOptions = (function () {
                function SharedOptions() {
                    this.allOptions = {};
                }
                SharedOptions.prototype.getOption = function (id) {
                    return this.allOptions[id];
                };
                SharedOptions.prototype.setOption = function (id, obj) {
                    this.allOptions[id] = {
                        id: id,
                        settings: obj.settings,
                        option: obj.option,
                        default: obj.default
                    };
                };
                SharedOptions.prototype.dispose = function () {
                    this.allOptions = {};
                };
                SharedOptions = __decorate([
                    aurelia_dependency_injection_1.singleton()
                ], SharedOptions);
                return SharedOptions;
            }());
            exports_1("SharedOptions", SharedOptions);
        }
    };
});
