// aurelia-toolbelt/dist/amd/components/jquery/lazy-image/aut-lazy-image.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework", "jquery", "jquery-lazy"], function (require, exports, aurelia_framework_1, $) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var JQueryLazyImage = (function () {
        function JQueryLazyImage(element) {
            this.element = element;
            this.backgroundMode = false;
            this.name = 'lazy';
            this.chainable = true;
            this.autoDestroy = true;
            this.bind = 'load';
            this.threshold = 500;
            this.visibleOnly = false;
            this.appendScroll = window;
            this.scrollDirection = 'both';
            this.imageBase = null;
            this.defaultImage = '';
            this.placeholder = null;
            this.delay = -1;
            this.combined = false;
            this.effect = 'show';
            this.effectTime = 0;
            this.enableThrottle = true;
            this.throttle = 250;
        }
        JQueryLazyImage.prototype.attached = function () {
            this.backgroundMode = this.backgroundMode === true || this.backgroundMode === 'true';
            if (this.customLoader !== undefined) {
                $(this.element.previousElementSibling).attr('data-loader', 'customLoader');
            }
            var config = {
                customLoader: this.customLoader,
                name: this.name,
                chainable: this.chainable,
                autoDestroy: this.autoDestroy,
                bind: this.bind,
                threshold: this.threshold,
                visibleOnly: this.visibleOnly,
                appendScroll: this.appendScroll,
                scrollDirection: this.scrollDirection,
                imageBase: this.imageBase,
                defaultImage: this.defaultImage,
                placeholder: this.placeholder,
                delay: this.delay,
                combined: this.combined,
                effect: this.effect,
                effectTime: this.effectTime,
                enableThrottle: this.enableThrottle,
                throttle: this.throttle,
                beforeLoad: this.beforeLoad,
                afterLoad: this.afterLoad,
                onError: this.error
            };
            $(this.element.previousElementSibling).lazy(config);
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], JQueryLazyImage.prototype, "backgroundMode", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "style", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "class", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "url", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "name", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Boolean)
        ], JQueryLazyImage.prototype, "chainable", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Boolean)
        ], JQueryLazyImage.prototype, "autoDestroy", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "bind", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Number)
        ], JQueryLazyImage.prototype, "threshold", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Boolean)
        ], JQueryLazyImage.prototype, "visibleOnly", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], JQueryLazyImage.prototype, "appendScroll", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "scrollDirection", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "imageBase", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "defaultImage", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "placeholder", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Number)
        ], JQueryLazyImage.prototype, "delay", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Boolean)
        ], JQueryLazyImage.prototype, "combined", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], JQueryLazyImage.prototype, "effect", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Number)
        ], JQueryLazyImage.prototype, "effectTime", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Boolean)
        ], JQueryLazyImage.prototype, "enableThrottle", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Number)
        ], JQueryLazyImage.prototype, "throttle", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], JQueryLazyImage.prototype, "beforeLoad", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], JQueryLazyImage.prototype, "afterLoad", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], JQueryLazyImage.prototype, "error", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], JQueryLazyImage.prototype, "customLoader", void 0);
        JQueryLazyImage = __decorate([
            aurelia_framework_1.inject(Element),
            aurelia_framework_1.containerless(),
            aurelia_framework_1.customElement('aut-lazy-image'),
            __metadata("design:paramtypes", [Element])
        ], JQueryLazyImage);
        return JQueryLazyImage;
    }());
    exports.JQueryLazyImage = JQueryLazyImage;
});
