// aurelia-toolbelt/dist/system/components/jquery/metis-menu/aut-metis-menu.js
System.register(["aurelia-framework", "jquery", "metismenu"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, $, JQueryMetisMenu;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function ($_1) {
                $ = $_1;
            },
            function (_1) {
            }
        ],
        execute: function () {
            JQueryMetisMenu = (function () {
                function JQueryMetisMenu(element) {
                    this.element = element;
                    this.class = '';
                    this.style = '';
                }
                JQueryMetisMenu.prototype.attached = function () {
                    var _this = this;
                    $(this.metismenu).metisMenu()
                        .on('show.metismenu', function (event) {
                        console.log("show menu: " + JSON.stringify(event));
                        console.log(_this.showMenu);
                        var localEvent = _this.showMenu;
                        if (localEvent !== null || localEvent !== undefined) {
                            Promise.resolve(function () {
                                localEvent(event);
                            });
                        }
                    })
                        .on('shown.metismenu', function (event) {
                        console.log("shown menu: " + JSON.stringify(event));
                        var localEvent = _this.shownMenu;
                        if (localEvent !== null || localEvent !== undefined) {
                            Promise.resolve(function () {
                                localEvent(event);
                            });
                        }
                    })
                        .on('hide.metismenu', function (event) {
                        console.log("hide menu: " + JSON.stringify(event));
                        var localEvent = _this.hideMenu;
                        if (localEvent !== null || localEvent !== undefined) {
                            Promise.resolve(function () {
                                localEvent(event);
                            });
                        }
                    })
                        .on('hidden.metismenu', function (event) {
                        console.log("menu hidden: " + JSON.stringify(event));
                        var localEvent = _this.hiddenMenu;
                        if (localEvent !== null || localEvent !== undefined) {
                            Promise.resolve(function () {
                                localEvent(event);
                            });
                        }
                    });
                };
                JQueryMetisMenu.prototype.detached = function () {
                    $(this.metismenu).metisMenu('dispose');
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], JQueryMetisMenu.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], JQueryMetisMenu.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], JQueryMetisMenu.prototype, "showMenu", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], JQueryMetisMenu.prototype, "shownMenu", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], JQueryMetisMenu.prototype, "hideMenu", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], JQueryMetisMenu.prototype, "hiddenMenu", void 0);
                JQueryMetisMenu = __decorate([
                    aurelia_framework_1.inject(Element),
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('aut-metis-menu'),
                    __metadata("design:paramtypes", [Element])
                ], JQueryMetisMenu);
                return JQueryMetisMenu;
            }());
            exports_1("JQueryMetisMenu", JQueryMetisMenu);
        }
    };
});
