// aurelia-toolbelt/dist/system/custom-attributes/vanilla/masked/aut-masked.js
System.register(["aurelia-framework", "aurelia-dependency-injection"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_dependency_injection_1, InputMask, MaskedInputCustomAttribute;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_dependency_injection_1_1) {
                aurelia_dependency_injection_1 = aurelia_dependency_injection_1_1;
            }
        ],
        execute: function () {
            InputMask = require('inputmask');
            MaskedInputCustomAttribute = (function () {
                function MaskedInputCustomAttribute(element) {
                    this.element = element;
                }
                MaskedInputCustomAttribute.prototype.maskChanged = function (newMask) {
                    var im = new InputMask(newMask);
                    im.mask(this.element);
                };
                MaskedInputCustomAttribute.prototype.regexChanged = function (newRegex) {
                    var im = new InputMask({ regex: newRegex });
                    im.mask(this.element);
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay, primaryProperty: true }),
                    __metadata("design:type", String)
                ], MaskedInputCustomAttribute.prototype, "mask", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], MaskedInputCustomAttribute.prototype, "regex", void 0);
                MaskedInputCustomAttribute = __decorate([
                    aurelia_dependency_injection_1.inject(Element),
                    aurelia_framework_1.customAttribute('aut-masked'),
                    __metadata("design:paramtypes", [Element])
                ], MaskedInputCustomAttribute);
                return MaskedInputCustomAttribute;
            }());
            exports_1("MaskedInputCustomAttribute", MaskedInputCustomAttribute);
        }
    };
});
