// aurelia-toolbelt/dist/commonjs/components/vanilla/divider/aut-divider.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var sharedIndex_1 = require("./../../../utilities/vanilla/sharedIndex");
var aurelia_framework_1 = require("aurelia-framework");
var DividerCustomElement = (function () {
    function DividerCustomElement(element, sharedIndex) {
        this.element = element;
        this.sharedIndex = sharedIndex;
        this.vertical = false;
        this.color = '#b5b5b5';
        this.backgroundColor = '#fff';
        this.lineColor = '#dbdbdb';
    }
    DividerCustomElement.prototype.attached = function () {
        this.vertical = (this.vertical === '' && this.element.hasAttribute('vertical')) || this.vertical.toString() === 'true';
        var value = this.divider.innerText;
        this.divider.innerText = '';
        this.divider.setAttribute('data-content', value);
        var css = ".is-divider-vertical[data-content]::after,.is-divider[data-content]::after{\n      background:" + this.backgroundColor + "!important;color:" + this.color + "!important}\n      .is-divider{border-top:.1rem solid " + this.lineColor + "!important;}\n      .is-divider-vertical::before{border-left:.1rem solid " + this.lineColor + "!important;}\n      ";
        aurelia_framework_1.DOM.injectStyles(css, null, null, 'aut_divider_injected_style');
    };
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
        __metadata("design:type", String)
    ], DividerCustomElement.prototype, "id", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], DividerCustomElement.prototype, "class", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", Object)
    ], DividerCustomElement.prototype, "vertical", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], DividerCustomElement.prototype, "color", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], DividerCustomElement.prototype, "backgroundColor", void 0);
    __decorate([
        aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
        __metadata("design:type", String)
    ], DividerCustomElement.prototype, "lineColor", void 0);
    DividerCustomElement = __decorate([
        aurelia_framework_1.customElement('aut-divider'),
        aurelia_framework_1.inject(Element, sharedIndex_1.SharedIndex),
        aurelia_framework_1.containerless(),
        __metadata("design:paramtypes", [Element, sharedIndex_1.SharedIndex])
    ], DividerCustomElement);
    return DividerCustomElement;
}());
exports.DividerCustomElement = DividerCustomElement;
