// aurelia-toolbelt/dist/system/components/vanilla/scrollup/aut-scrollup.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, ScrollUpCustomElement;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            ScrollUpCustomElement = (function () {
                function ScrollUpCustomElement() {
                    this.threshold = 150;
                }
                ScrollUpCustomElement.prototype.checkScrollTop = function () {
                    if (this.threshold <= 0) {
                        this.threshold = 0;
                    }
                    if (document.body.scrollTop > this.threshold || document.documentElement.scrollTop > this.threshold) {
                        this.scrollupButton.style.display = 'block';
                        if (this.shownScrollUp) {
                            this.shownScrollUp();
                        }
                    }
                    else {
                        this.scrollupButton.style.display = 'none';
                        if (this.hideScrollUp) {
                            this.hideScrollUp();
                        }
                    }
                };
                ScrollUpCustomElement.prototype.goToUp = function () {
                    if (this.beforeScrollUp) {
                        this.beforeScrollUp();
                    }
                    if (!document.body.scroll) {
                        window.scrollTo(0, 0);
                        if (this.afterScrollUp) {
                            this.afterScrollUp();
                        }
                        return;
                    }
                    document.body.scroll({
                        top: 0,
                        left: 0,
                        behavior: 'smooth'
                    });
                    document.documentElement.scroll({
                        top: 0,
                        left: 0,
                        behavior: 'smooth'
                    });
                    if (this.afterScrollUp) {
                        this.afterScrollUp();
                    }
                };
                ScrollUpCustomElement.prototype.attached = function () {
                    var _this = this;
                    this.threshold = Number(this.threshold);
                    window.onscroll = function () { return _this.checkScrollTop(); };
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], ScrollUpCustomElement.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], ScrollUpCustomElement.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], ScrollUpCustomElement.prototype, "threshold", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], ScrollUpCustomElement.prototype, "beforeScrollUp", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], ScrollUpCustomElement.prototype, "afterScrollUp", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], ScrollUpCustomElement.prototype, "shownScrollUp", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], ScrollUpCustomElement.prototype, "hideScrollUp", void 0);
                ScrollUpCustomElement = __decorate([
                    aurelia_framework_1.customElement('aut-scrollup')
                ], ScrollUpCustomElement);
                return ScrollUpCustomElement;
            }());
            exports_1("ScrollUpCustomElement", ScrollUpCustomElement);
        }
    };
});
