// aurelia-toolbelt/dist/system/components/bootstrap/alert/abt-alert.js
System.register(["aurelia-framework", "aurelia-dependency-injection", "jquery"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    };
    var __generator = (this && this.__generator) || function (thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = y[op[0] & 2 ? "return" : op[0] ? "throw" : "next"]) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [0, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, aurelia_dependency_injection_1, $, BootstrapAlert;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            },
            function (aurelia_dependency_injection_1_1) {
                aurelia_dependency_injection_1 = aurelia_dependency_injection_1_1;
            },
            function ($_1) {
                $ = $_1;
            }
        ],
        execute: function () {
            BootstrapAlert = (function () {
                function BootstrapAlert(element) {
                    this.element = element;
                    this.size = 'md';
                    this.type = 'primary';
                    this.animate = true;
                    this.style = '';
                    this.class = '';
                    this.showAlert = null;
                    this.dismissible = false;
                }
                BootstrapAlert.prototype.attached = function () {
                    var _this = this;
                    var onlyDismissibleAttribute = (this.dismissible === '' && this.element.hasAttribute('dismissible'));
                    this.dismissible = onlyDismissibleAttribute || this.dismissible.toString() === 'true';
                    var onlyAnimateAttribute = (this.animate === '' && this.element.hasAttribute('animate'));
                    this.animate = onlyAnimateAttribute || this.animate === 'true' || this.animate === true;
                    if (this.bsClose) {
                        $(alert).on('close.bs.alert', function () {
                            if (_this.bsClose) {
                                _this.bsClose();
                            }
                        });
                    }
                    if (this.bsClosed) {
                        $(alert).on('closed.bs.alert', function () {
                            if (_this.bsClosed) {
                                _this.bsClosed();
                            }
                        });
                    }
                };
                BootstrapAlert.prototype.showAlertChanged = function (newValue) {
                    return __awaiter(this, void 0, void 0, function () {
                        var continueShow, continueHide;
                        return __generator(this, function (_a) {
                            switch (_a.label) {
                                case 0:
                                    if (!newValue) return [3, 3];
                                    continueShow = true;
                                    if (!this.bsShow) return [3, 2];
                                    return [4, this.bsShow({ target: this.alert })];
                                case 1:
                                    continueShow = (_a.sent());
                                    _a.label = 2;
                                case 2:
                                    continueShow = continueShow === undefined || continueShow === null ? true : continueShow;
                                    if (!continueShow) {
                                        this.showAlert = !newValue;
                                        return [2];
                                    }
                                    if (this.animate) {
                                        $(this.alert).fadeIn();
                                    }
                                    else {
                                        $(this.alert).show();
                                    }
                                    if (this.bsShown) {
                                        this.bsShown({ target: this.alert });
                                    }
                                    return [3, 6];
                                case 3:
                                    continueHide = true;
                                    if (!this.bsHide) return [3, 5];
                                    return [4, this.bsHide({ target: this.alert })];
                                case 4:
                                    continueHide = (_a.sent());
                                    _a.label = 5;
                                case 5:
                                    continueHide = continueHide === undefined || continueHide === null ? true : continueHide;
                                    if (!continueHide) {
                                        this.showAlert = !newValue;
                                        return [2];
                                    }
                                    if (this.animate) {
                                        $(this.alert).fadeOut();
                                    }
                                    else {
                                        $(this.alert).hide();
                                    }
                                    if (this.bsHidden) {
                                        this.bsHidden({ target: this.alert });
                                    }
                                    _a.label = 6;
                                case 6: return [2];
                            }
                        });
                    });
                };
                BootstrapAlert.prototype.detached = function () {
                    $(this.alert).alert('close');
                    $(this.alert).alert('dispose');
                };
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapAlert.prototype, "size", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapAlert.prototype, "type", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", Object)
                ], BootstrapAlert.prototype, "animate", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapAlert.prototype, "style", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapAlert.prototype, "class", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Boolean)
                ], BootstrapAlert.prototype, "showAlert", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", Object)
                ], BootstrapAlert.prototype, "dismissible", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapAlert.prototype, "bsShow", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapAlert.prototype, "bsShown", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapAlert.prototype, "bsHide", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapAlert.prototype, "bsHidden", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapAlert.prototype, "bsClose", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Function)
                ], BootstrapAlert.prototype, "bsClosed", void 0);
                BootstrapAlert = __decorate([
                    aurelia_dependency_injection_1.inject(Element),
                    aurelia_framework_1.customElement('abt-alert'),
                    __metadata("design:paramtypes", [Element])
                ], BootstrapAlert);
                return BootstrapAlert;
            }());
            exports_1("BootstrapAlert", BootstrapAlert);
        }
    };
});
