// aurelia-toolbelt/dist/system/components/bootstrap/alert/abt-alert-link.js
System.register(["aurelia-framework"], function (exports_1, context_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var aurelia_framework_1, BootstrapAlertLink;
    return {
        setters: [
            function (aurelia_framework_1_1) {
                aurelia_framework_1 = aurelia_framework_1_1;
            }
        ],
        execute: function () {
            BootstrapAlertLink = (function () {
                function BootstrapAlertLink() {
                    this.id = '';
                    this.href = '';
                    this.disabled = false;
                }
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneTime }),
                    __metadata("design:type", String)
                ], BootstrapAlertLink.prototype, "id", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
                    __metadata("design:type", String)
                ], BootstrapAlertLink.prototype, "href", void 0);
                __decorate([
                    aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
                    __metadata("design:type", Object)
                ], BootstrapAlertLink.prototype, "disabled", void 0);
                BootstrapAlertLink = __decorate([
                    aurelia_framework_1.inject(Element),
                    aurelia_framework_1.containerless(),
                    aurelia_framework_1.customElement('abt-alert-link')
                ], BootstrapAlertLink);
                return BootstrapAlertLink;
            }());
            exports_1("BootstrapAlertLink", BootstrapAlertLink);
        }
    };
});
