// aurelia-toolbelt/dist/amd/components/bootstrap/popover/abt-popover.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "aurelia-framework", "jquery"], function (require, exports, aurelia_framework_1, $) {
    Object.defineProperty(exports, "__esModule", { value: true });
    var BootstrapPopoverCustomElement = (function () {
        function BootstrapPopoverCustomElement() {
            this.animation = true;
            this.container = false;
            this.delay = 0;
            this.html = false;
            this.placement = 'right';
            this.selector = false;
            this.title = '';
            this.trigger = 'click';
            this.offset = 0;
            this.fallbackPlacement = 'flip';
            this.boundary = 'scrollParent';
            this.template = '<div class="popover" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>';
        }
        BootstrapPopoverCustomElement.prototype.attached = function () {
            var _this = this;
            this.parentElement = this.popover.parentElement;
            var slotContent = this.html ? this.popover.innerHTML : this.popover.textContent;
            this.offset = Number(this.offset);
            this.animation = (this.animation === '' && this.popoverTemplate.hasAttribute('animation')) || this.animation.toString() === 'true';
            this.container = (this.container === '' && this.popoverTemplate.hasAttribute('container')) || this.container.toString() === 'true';
            this.html = (this.html === '' && this.popoverTemplate.hasAttribute('html')) || this.html.toString() === 'true';
            this.selector = (this.selector === '' && this.popoverTemplate.hasAttribute('selector')) || this.selector.toString() === 'true';
            $(this.parentElement).popover({
                'content': slotContent,
                'title': this.title,
                'html': this.html,
                'template': this.template,
                'animation': this.animation,
                'placement': this.placement,
                'container': this.container,
                'delay': this.delay,
                'trigger': this.trigger,
                'offset': this.offset,
                'fallbackPlacement': this.fallbackPlacement,
                'boundary': this.boundary
            });
            this.popover.remove();
            if (this.bsShow) {
                $(this.parentElement).on('show.bs.popover', function () {
                    if (_this.bsShow) {
                        _this.bsShow();
                    }
                });
            }
            if (this.bsShown) {
                $(this.parentElement).on('shown.bs.popover', function () {
                    if (_this.bsShown) {
                        _this.bsShown();
                    }
                });
            }
            if (this.bsHide) {
                $(this.parentElement).on('hide.bs.popover', function () {
                    if (_this.bsHide) {
                        _this.bsHide();
                    }
                });
            }
            if (this.bsHidden) {
                $(this.parentElement).on('hidden.bs.popover', function () {
                    if (_this.bsHidden) {
                        _this.bsHidden();
                    }
                });
            }
            if (this.bsInserted) {
                $(this.parentElement).on('inserted.bs.popover', function () {
                    if (_this.bsInserted) {
                        _this.bsInserted();
                    }
                });
            }
        };
        BootstrapPopoverCustomElement.prototype.detached = function () {
            $(this.parentElement).popover('dispose');
        };
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "animation", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "container", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "delay", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "html", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "placement", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "selector", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "title", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapPopoverCustomElement.prototype, "trigger", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "offset", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", Object)
        ], BootstrapPopoverCustomElement.prototype, "fallbackPlacement", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapPopoverCustomElement.prototype, "boundary", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.oneWay }),
            __metadata("design:type", String)
        ], BootstrapPopoverCustomElement.prototype, "template", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapPopoverCustomElement.prototype, "bsShow", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapPopoverCustomElement.prototype, "bsShown", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapPopoverCustomElement.prototype, "bsHide", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapPopoverCustomElement.prototype, "bsHidden", void 0);
        __decorate([
            aurelia_framework_1.bindable({ defaultBindingMode: aurelia_framework_1.bindingMode.twoWay }),
            __metadata("design:type", Function)
        ], BootstrapPopoverCustomElement.prototype, "bsInserted", void 0);
        BootstrapPopoverCustomElement = __decorate([
            aurelia_framework_1.containerless(),
            aurelia_framework_1.customElement('abt-popover')
        ], BootstrapPopoverCustomElement);
        return BootstrapPopoverCustomElement;
    }());
    exports.BootstrapPopoverCustomElement = BootstrapPopoverCustomElement;
});
